src/Courses/TopicGrades/GradeInfoHeader.tsx

```diff
@@ -120,10 +120,22 @@ export const GradeInfoHeader: React.FC<GradeInfoHeaderProps> = ({
                         } else {
                             // there should only be one studentGradeId in this case...
                             (map[wb.studentGradeId] || (map[wb.studentGradeId] = [])).push(parseInt(id, 10));
-                            if (wb.studentGradeId !== grade.id) logger.error(`Workbooks for grade ${grade.id} are out of sync with the corresponding grade object.`);
+                            if (wb.studentGradeId !== grade.id) {
+                                logger.error(`Workbooks for grade ${grade.id} are out of sync with the corresponding grade object.`);
+                            }
                         }
                         return map;
                     }, {} as { [k: number]: number[] });
+
+                    // In this case, the latest version has not been submitted yet.
+                    const currentVersionMapKeys = _.keys(currentVersionMap);
+                    if (grade.gradeInstances && currentVersionMapKeys.length === grade.gradeInstances.length - 1) {
```

Comment:
either we should make this a `>` or add a check that if it is more than 1 off we console.warn
This way we don't get silent bugs that are hard to find